### ISSUE TITLE
[D3D11]: Wrong input layout used on equal vertex layouts but different vertex shaders

### ISSUE THREAD

**Issue by Helco** [mellinoe/veldrid#294]
When the same vertex layout is used for multiple vertex shaders, the caching of InputLayout instance by the d3d11 backend causes various bugs which usually manifests in nothing being drawn or vertex attributes being zero when processed by the shader.

This is most likely because the InputLayout references the vertex shader bytecode? So when two equal vertex layouts are used the same InputLayoutCacheKey points an old InputLayout with a different vertex shader. The equality of InputLayoutCacheKey probably should compare against that bytecode as well as the vertex layout.

An easy workaround exists by e.g. using a unique name in any of the vertex layout element names, ensuring that no two vertex layouts are equal.

**Comment by mellinoe** [comment]
Yes, this is definitely a problem. The solution is just as you've said: InputLayoutCacheKey needs to consider the bytecode array in its equality and GetHashCode methods. I don't have the bandwidth to fix this at the moment, but please feel free to submit a pull request if you are willing.
